src/scripts/apploader.js

```diff
@@ -29,6 +29,16 @@
         );
     }
 
+    try {
+        Promise.resolve();
+    } catch (ex) {
+        // this checks for several cases actually, typical is
+        // Promise() being missing on some legacy browser, and a funky one
+        // is Promise() present but buggy on WebOS 2
+        window.Promise = undefined;
+        self.Promise = undefined;
+    }
+
     if (!self.Promise) {
```

Comment:
Can you also remove this condition while you're in there?

The promise polyfill already does that check, so no need to repeat it :)

Reply:
Too late, I'm already merging this to keep open PRs down :) just add it to an unrelated PR and no one will notice.